rewrite-maven/src/main/java/org/openrewrite/maven/tree/MavenRepository.java

```diff
@@ -133,6 +133,17 @@ public MavenRepository(
         this.deriveMetadataIfMissing = deriveMetadataIfMissing;
     }
 
+    public static MavenRepository from(MavenRepository repository, String url) {
+        return new MavenRepository(
+                repository.getId(),
+                url,
+                repository.getReleases(),
+                repository.getSnapshots(),
+                repository.getUsername(),
+                repository.getPassword(),
+                repository.getTimeout());
+    }
+
```

Comment:
Use the existing `MavenRepository.withUri()` instead of creating this method.

Reply:
will do. I added this because there are 2 nonFinal fields which are then copied. but since we dont care about `knownToExist` it's only `deriveMetadataIfMissing` and I dont think that one matters either?